saltgui/static/scripts/output.js

```diff
@@ -21,6 +21,63 @@
 
 class Output {
 
+  // format a JSON object
+  // based on an initial indentation and an indentation increment
+  static _formatJSON0(value, indentStr, indentCnt) {
+
+    if(typeof value !== "object") {
+      // a simple type
+      // leave that to the builtin function
+      return JSON.stringify(value);
+    }
+  
+    if(Array.isArray(value)) {
+      // an array
+      // put each element on its own line
+      let str = "[";
+      let separator = "";
+      for(const elem of value) {
+        str += separator + "\n" + indentStr + indentCnt +
+          Output._formatJSON0(elem, indentStr + indentCnt, indentCnt);
+        separator = ",";
+      }
+      if(value.length === 0) {
+        // show the brackets for an empty array a bit wider apart
+        str += " ";
+      } else {
+        str += "\n" + indentStr;
+      }
+      str += "]";
+      return str;
+    }
```

Comment:
I don't see any sorting applied here so why not just
```json
JSON.stringify([1,2,3,4,5], null, indent)
"[
    1,
    2,
    3,
    4,
    5
]"
```

Reply:
because the one of the values of array may be an object which needs sorting